heartbeat/iface-macvlan

```diff
@@ -0,0 +1,363 @@
+
+Expects to have a fully populated OCF RA-compliant environment set.
+END
+}
+
+macvlan_meta_data() {
+	cat <<END
+<?xml version="1.0"?>
+<!DOCTYPE resource-agent SYSTEM "ra-api-1.dtd">
+<resource-agent name="iface-macvlan" version="1.0">
+  <version>1.0</version>
+
+  <longdesc lang="en">
+    This resource manages MACVLAN network interfaces.
+    It can add, remove, configure MACVLANs.
+  </longdesc>
+
+  <shortdesc lang="en">
+    Manages MACVLAN network interfaces.
+  </shortdesc>
+
+  <parameters>
+    <parameter name="device" unique="0" required="1">
+      <longdesc lang="en">
+        Define the interface where MACVLAN should be attached.
+      </longdesc>
+      <shortdesc lang="en">
+        Network interface.
+      </shortdesc>
+      <content type="string"/>
+    </parameter>
+
+    <parameter name="name" unique="0" required="1">
+      <longdesc lang="en">
+        Define the MACVLAN NAME. It has to be a valid interface name (max 15 characters).
+      </longdesc>
+      <shortdesc lang="en">
+        Define the MACVLAN NAME.
+      </shortdesc>
+      <content type="string"/>
+    </parameter>
+
+    <parameter name="mode" unique="1">
+      <longdesc lang="en">
+        Define the name of the MACVLAN mode (currently only bridge is supported).
+      </longdesc>
+      <shortdesc lang="en">
+        Mode of the macvlan.
+      </shortdesc>
+      <content type="string" default="${OCF_RESKEY_mode_default}" />
+    </parameter>
+
+    <parameter name="mac">
+     <longdesc lang="en">
+      Set the interface MAC address explicitly.
+     </longdesc>
+     <shortdesc lang="en">MAC address of the macvlan</shortdesc>
+     <content type="string" />
+</parameter>
+  </parameters>
+
+  <actions>
+    <action name="start"        timeout="30s" />
+    <action name="stop"         timeout="20s" />
+    <action name="status"       timeout="20s" depth="0" interval="10s" />
+    <action name="monitor"      timeout="20s" depth="0" interval="10s" />
+    <action name="meta-data"    timeout="5s" />
+    <action name="validate-all" timeout="20s" />
+  </actions>
+</resource-agent>
+END
+}
+
+# check if the interface is admin up/down
+
+iface_is_up() {
+	if ! $IP2UTIL -o link show $1 | \
+	    sed -e 's#.*<##g' -e 's#>.*##' -e 's#LOWER_UP##g' | \
+	    grep -q UP; then
+		return 1
+	fi
+	return 0
+}
+
+# check if the slaves have link layer up/down
+# see kernel network documentation on meaning of LOWER_UP flag
+# for more in depth explanation on how it works
+# NOTE: this check is not reliable in virt environment
+# since interfaces are always LOWER_UP. There is no way
+# from the guest to know if the host has disconnected somehow
+
+iface_lower_is_up() {
+	if ! $IP2UTIL -o link show $1 | \
+	    grep -q LOWER_UP; then
+		return 1
+	fi
+	return 0
+}
+
+macvlan_validate() {
+	check_binary $IP2UTIL
+
+	if [ -z "$OCF_RESKEY_device" ]; then
+		ocf_log err "Invalid OCF_RESKEY_device: value cannot be empty"
```

Comment:
You should remove OCF_RESKEY_ in the errors, as that's appended by Pacemaker when you set e.g. device for the resource. OCF_RESKEY_ names are only visible to the users if they manually test it outside of Pacemaker.

And you should `return $OCF_ERR_CONFIGURED`. See <https://github.com/ClusterLabs/resource-agents/blob/main/doc/dev-guides/ra-dev-guide.asc#return-codes> for description of the return codes.

Reply:
I have checked that and now the name of the variable is reported correctly and the return codes are changed.